app.py

```diff
@@ -51,6 +98,69 @@ def get_memory():
 def actualise_cpu_data():
     cpu_service.calculate_cpu_usage()
 
+@app.route('/api/network')
+def network_monitoring_test():
+    # Get network I/O counters
+    try:
+        network_io_counters = psutil.net_io_counters()#return tuple that contains network I/O statistics for different network interfaces
+        bytes_sent = network_io_counters.bytes_sent
+        bytes_received = network_io_counters.bytes_recv
+
+    #perform some network operations
+        def network_operation(url):
+                try:
+                    response = requests.get(url)
+                    response.raise_for_status() #raise an exception for any http error status code
+                    logger.info("Network Operation succesful!")
+                    logger.info("Response status code: %s", response.status_code)
+                    def append_file(output_file):
+                        subprocess.run(logger.info("Response content: ", response.text), shell= True, text= True, capture_output= True, append= True, check= True, stdout=open(output_file))
+                    output_file="output.txt"
+                    file = open(output_file, 'w')
+                    output = append_file
+                    print (output, file=file)
+                    file.close()  
+
+                except requests.exceptions.RequestException as e:
+                    logger.error("Network operation failed: ", e)
+        url = "https://www.google.com"
+        network_operation(url)
+
+    #get updated network I/O counters
+        network_io_counters = psutil.net_io_counters() 
+        new_bytes_sent = network_io_counters.bytes_sent
+        new_bytes_received = network_io_counters.bytes_recv
+
+    #Calculate the data transferred during the network operation
+        data_sent = new_bytes_sent - bytes_sent
+        data_received = new_bytes_received -bytes_received
+
+    #Validate network monitoring
+        if data_sent > 0:
+            logger.info("Data sent during the network operation: %s", data_sent)
+        else:
+            logger.info("No data sent during the network operation")
+
+        if data_received > 0:
+            logger.info("Data received during the network operation: %s", data_received)
+        else:
+            logger.info("No data received during the network operation")
+    except Exception as e:
+        logger.error("An error occurred during network monitoring: %s", e)
+
+@app.route('/api/network/<hostname>')
+def api_network(hostname):
```

Comment:
Because it's an API Method (because of the `@app.route...`), this method should return something. Returning `print` is not a good thing.